ProcessSmarterTestPackage/PostProcessors/FormPartitionPostProcessor.cs

```diff
@@ -16,23 +16,101 @@ public override IList<ValidationError> GenerateErrors()
 
             var partitionIdentifier = Processor.ChildNodeWithName("identifier").ValueForAttribute("uniqueid");
             var itemGroups = Processor.ChildNodesWithName("itemgroup").ToList();
+            const string groupIdPattern = @"(\d+-\d+):([G|I])-(\d+-\d+)";
+
             foreach (var itemGroup in itemGroups)
             {
-                if (!itemGroup.ChildNodeWithName("identifier")
-                    .ValueForAttribute("uniqueid").Split(':').First()
-                    .Equals(partitionIdentifier, StringComparison.OrdinalIgnoreCase))
+                var match = Regex.Match(itemGroup.ChildNodeWithName("identifier")
+                    .ValueForAttribute("uniqueid"), groupIdPattern);
+                if (!match.Success)
                 {
                     result.Add(new ValidationError
                     {
-                        ErrorSeverity = ErrorSeverity.Benign,
+                        ErrorSeverity = ErrorSeverity.Severe,
                         GeneratedMessage =
-                            $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} does not follow expected pattern {partitionIdentifier}:G-<ID>]",
+                            $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} does not follow expected pattern {groupIdPattern}]",
                         Key = "uniqueid",
                         ItemId = itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid"),
                         PackageType = PackageType,
                         Location = "itemgroup/identifier"
                     });
                 }
+                else
+                {
+                    if (!match.Groups[1].Value.Equals(partitionIdentifier))
+                    {
+                        result.Add(new ValidationError
+                        {
+                            ErrorSeverity = ErrorSeverity.Benign,
+                            GeneratedMessage =
+                                $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} first portion {match.Groups[1]} does not match partition ID and expected value {partitionIdentifier}]",
+                            Key = "uniqueid",
+                            ItemId = itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid"),
+                            PackageType = PackageType,
+                            Location = "itemgroup/identifier"
+                        });
+                    }
+                    if (match.Groups[2].Value.Equals("G"))
+                    {
+                        if (itemGroup.ChildNodesWithName("passageref").Count() != 1)
```

Comment:
"G-" items that do not contain a passageref that matches a valid stimuli fail at runtime.